instrumentation/aws-lambda/aws-lambda-events-2.2/library/src/main/java/io/opentelemetry/instrumentation/awslambdaevents/v2_2/internal/AwsLambdaSqsInstrumenterFactory.java

```diff
@@ -39,10 +40,11 @@ public static Instrumenter<SQSMessage, Void> forMessage(OpenTelemetry openTeleme
 
   private static String spanName(SQSEvent event) {
     String source = "multiple_sources";
-    if (!event.getRecords().isEmpty()) {
-      String messageSource = event.getRecords().get(0).getEventSource();
-      for (int i = 1; i < event.getRecords().size(); i++) {
-        SQSMessage message = event.getRecords().get(i);
+    List<SQSMessage> records = event.getRecords();
+    if ((records != null) && !records.isEmpty()) {
```

Comment:
```suggestion
    if (records != null && !records.isEmpty()) {
```

Reply:
In my opinion in this case the parentheses don't really help because there is no way to place the parenthesis in expression to get a different order of operations. If this expression had a mix of && and || I'd totally agree with adding the parenthesis.

Reply:
There are languages in which this is valid syntax:

```
if (records != (null && !records.isEmpty())) {
```

My point is that a reader who sees the bare expression without clarifying parens has to think about it. The parens take out that thinking.